EpsilonLights/Src/Lights.c

```diff
@@ -7,90 +7,89 @@ void updateLightsTask(void const* arg)
-        headlightsLow = (lightsInputs >> HLOW_INPUT_INDEX) & 1;
-        headlightsHigh = (lightsInputs >> HHIGH_INPUT_INDEX) & 1;
-        rightSignal = (lightsInputs >> RSIGNAL_INPUT_INDEX) & 1;
-        leftSignal = (lightsInputs >> LSIGNAL_INPUT_INDEX) & 1;
-        hazards = (lightsInputs >> HAZARDS_INPUT_INDEX) & 1;
-        bmsStrobe = ((auxBmsInputs[1] >> 0) & STROBE_FAULT_MASK) & 1;
-        regenBrakeInt |= (driversInputs[1] & REGEN_BRAKE_MASK_1) >> 4;
-        regenBrakeInt |= (driversInputs[2] & REGEN_BRAKE_MASK_2) << 4;
-
-        /* UPDATE HEADLIGHTS */
-        if ((headlightsOff))
-        {
-            HAL_GPIO_WritePin(HHIGH_GPIO_Port, HHIGH_Pin, LIGHT_OFF);
-            HAL_GPIO_WritePin(HLOW_GPIO_Port, HLOW_Pin, LIGHT_OFF);
-        }
-        else if ((headlightsLow && headlightsHigh))
-        {
-            // Error state, turn only the low headlights on.
-            HAL_GPIO_WritePin(HHIGH_GPIO_Port, HHIGH_Pin, LIGHT_OFF);
-            HAL_GPIO_WritePin(HLOW_GPIO_Port, HLOW_Pin, LIGHT_ON);
-        }
-        else
-        {
-            HAL_GPIO_WritePin(HHIGH_GPIO_Port, HHIGH_Pin, headlightsHigh);
-            HAL_GPIO_WritePin(HLOW_GPIO_Port, HLOW_Pin, headlightsLow);
-        }
+        updateLights1 (&lightsCharacteristics, &prevWakeTime);
+    }
+}
 
-        /* UPDATE BRAKE LIGHTS */
-        brakes = (driversInputs[BRAKES_INPUT_INDEX_P1] >> BRAKES_INPUT_INDEX_P2) & 1;
-        regenBrakeFloat = (float)regenBrakeInt;
+void updateLights1 (lightsInfo* lightsCharacteristics, uint32_t* prevWakeTime)
+{
+    lightsCharacteristics -> regenBrakeInt = 0;
+    lightsCharacteristics -> regenBrakeFloat = 0;
 
-        if (brakes || regenBrakeFloat > NON_ZERO_THRESHOLD)
-        {
-            HAL_GPIO_WritePin(BRAKE_GPIO_Port, BRAKE_Pin, LIGHT_ON);
-        }
-        else
-        {
-            HAL_GPIO_WritePin(BRAKE_GPIO_Port, BRAKE_Pin, LIGHT_OFF);
-        }
+    osDelayUntil(prevWakeTime, LIGHTS_UPDATE_FREQ);
+    lightsCharacteristics -> headlightsOff = (lightsInputs >> HOFF_INPUT_INDEX) & 1; //0000 1111
+    lightsCharacteristics -> headlightsLow = (lightsInputs >> HLOW_INPUT_INDEX) & 1;
+    lightsCharacteristics -> headlightsHigh = (lightsInputs >> HHIGH_INPUT_INDEX) & 1;
+    lightsCharacteristics -> rightSignal = (lightsInputs >> RSIGNAL_INPUT_INDEX) & 1;
+    lightsCharacteristics -> leftSignal = (lightsInputs >> LSIGNAL_INPUT_INDEX) & 1;
+    lightsCharacteristics -> hazards = (lightsInputs >> HAZARDS_INPUT_INDEX) & 1;
+    lightsCharacteristics -> bmsStrobe = ((auxBmsInputs[1] >> 0) & STROBE_FAULT_MASK) & 1;
+    lightsCharacteristics -> regenBrakeInt |= (driversInputs[1] & REGEN_BRAKE_MASK_1) >> 4;// 1010 1111 & 1111 0000 = 1010 0000 // 0000 1010 // 0000 0000 | 0000 1010 = 0000 1010
+    lightsCharacteristics -> regenBrakeInt |= (driversInputs[2] & REGEN_BRAKE_MASK_2) << 4;// 1100 1011 & 1111 1111 = 1100 1011 // 1011 0000 // 0000 1010 | 1011 0000 = 1011 1010
```

Comment:
Seems like these comments were for you to check your unit tests so you should probably remove it if that's the case. Do the same everywhere else too.